src/cljs/harja/views/urakka/pot2/paallystekerros.cljs

```diff
@@ -55,123 +55,157 @@
+                                                                      :materiaali (pot2-tiedot/rivi->massa-tai-murske rivi {:massat massat})
+                                                                      :fmt :komponentti}]])))
+        :validoi [[:ei-tyhja-jos-toinen-avain-ei-joukossa :toimenpide [pot2-domain/+kulutuskerros-toimenpide-karhinta+] "Anna arvo"]]}
+       {:otsikko "Leveys (m)" :nimi :leveys :tyyppi :positiivinen-numero :tasaa :oikea
+        :tayta-alas? pot2-tiedot/tayta-alas?-fn :desimaalien-maara 2
+        :leveys (:perusleveys pot2-yhteiset/gridin-leveydet) :validoi [[:ei-tyhja "Anna arvo"]]
+        :validoi-kentta-fn (fn [numero] (v/validoi-numero numero 0 20 2))}
+       {:otsikko "Kok.m. (t)" :nimi :kokonaismassamaara :tyyppi :positiivinen-numero :tasaa :oikea
+        :leveys (:perusleveys pot2-yhteiset/gridin-leveydet) :validoi [[:ei-tyhja "Anna arvo"]]
+        :validoi-kentta-fn (fn [numero] (v/validoi-numero numero 0 1000000 1))}
+       {:otsikko "Pinta-ala (m²)" :nimi :pinta_ala :tyyppi :positiivinen-numero :tasaa :oikea :muokattava? (constantly false)
+        :fmt #(fmt/desimaaliluku-opt % 1)
+        :hae (fn [rivi]
+               (when-let [pituus (tr/laske-tien-pituus (into {}
+                                                         (map (juxt key (comp :pituus val)))
+                                                         (get tr-osien-pituudet (:tr-numero rivi)))
+                                   rivi)]
+                 (when (:leveys rivi)
+                   (* (:leveys rivi) pituus))))
+        :leveys (:perusleveys pot2-yhteiset/gridin-leveydet)}
+       {:otsikko "Massa\u00ADmenekki (kg/m\u00B2)" :nimi :massamenekki :tyyppi :positiivinen-numero :tasaa :oikea
+        :fmt #(fmt/desimaaliluku-opt % 1) :muokattava? (constantly false)
+        :hae (fn [rivi]
+               (paikkaus/massamaara-ja-pinta-ala->massamenekki
+                 (:kokonaismassamaara rivi)
+                 (:pinta_ala rivi)))
+        :tayta-alas? pot2-tiedot/tayta-alas?-fn :leveys (:perusleveys pot2-yhteiset/gridin-leveydet)}
+       {:otsikko "" :nimi :kulutuspaallyste-toiminnot :tyyppi :reagent-komponentti :leveys (:toiminnot pot2-yhteiset/gridin-leveydet)
+        :tasaa :keskita :komponentti-args [e! app kirjoitusoikeus? kohdeosat-atom :paallystekerros voi-muokata? ohjauskahva]
+        :komponentti pot2-yhteiset/rivin-toiminnot-sarake}]
+      kohdeosat-atom]
+
+     (let [kokpituus (reduce (fn [acc data]
+                               (when-let [pituus (tr/laske-tien-pituus (into {}
+                                                                         (map (juxt key (comp :pituus val)))
+                                                                         (get tr-osien-pituudet (:tr-numero (second data))))
+                                                   (second data))]
+                                 (+ acc pituus))) 0 @kohdeosat-atom)]
```

Comment:
0 @kohdeosat-atom - parametrit voisi erottaa omalle rivilleen, niin lukeminen helpottuisi :)

Reply:
Tehdään näin :+1: